jetty-webapp/src/main/java/org/eclipse/jetty/webapp/ClasspathPattern.java

```diff
@@ -717,30 +704,18 @@ public boolean match(String name, URL url)
         // Treat path elements as packages for name matching
         name=name.replace("/",".");
 
-        Boolean byName = _patterns.isIncludedAndNotExcluded(name);
-        if (Boolean.FALSE.equals(byName))
-            return byName; // Already excluded so no need to check location.
-        
-        // Try to find a file path for location matching
-        Boolean byLocation = null;
-        try
+        return  IncludeExcludeSet.or(_packageOrNamePatterns, name, _locations, ()->
```

Comment:
Too difficult to understand, see above.

Reply:
Refactor into a private method in this class and keep the original logic, but shortcutted as you've done in the new implementation.